### ISSUE TITLE
Errors in enums generated in inline schemas

### ISSUE THREAD

**Issue by msmiths** [smallrye/smallrye-open-api#403]
## Overview

Generating an inline schema for a class that contains a property whose type is an enum results in the generation of multiple refs in the OpenAPI document.  Both the enum definition and the property on the referencing class need to be annotated with the `@Schema` annotation.

You can see this in the `AirlinesAppTckTest`.

The **Gender** enum is defined as follows:

```
@Schema(enumeration = {"Male", "Female", "Other"})
public enum Gender {
    MALE, FEMALE, OTHER
}
```

The **User** class defines a `gender` property of type `Gender` that is also annotated with a `@Schema` annotation as follows:
```
    @Schema(required = true, example = "M")
    private Gender gender;
```
When the `User` class is referenced by a `@Schema` annotation on the `UserResource::createUser` method, it results in an inline schema being generated, but the definition of the `gender` property is as follows:
```
  gender:
    allOf:
    - $ref: '#/components/schemas/Gender'
    - example: M
    - $ref: '#/components/schemas/Gender'
    - example: M
    - $ref: '#/components/schemas/Gender'
    - example: M
    - $ref: '#/components/schemas/Gender'
    - example: M
```
The relevant annotation on the `UserResource::createUser` method is as follows:

```
    @RequestBody(
        ...
        content = @Content(
            mediaType = "application/json",
            schema = @Schema(
                name = "testUser",
                type = SchemaType.OBJECT,
                maxProperties = 1024,
                minProperties = 1,
                required = true,
                implementation = User.class
            ),
        ...
```

**Comment by MikeEdgar** [comment]
@msmiths - are you getting this behavior using `1.1.2` of the TCK also? Based on the output this looks like `2.0-RC2`. Just confirming. I do not get the repeating output myself with `1.1.2`.

**Comment by msmiths** [comment]
@MikeEdgar I am specifying the following in the parent POM:
```
        <version.eclipse.microprofile.openapi>2.0-SNAPSHOT</version.eclipse.microprofile.openapi>
```

**Comment by MikeEdgar** [comment]
@msmiths - which SmallRye branch are you using? I assume you have picked up @jmini's changes? I'd like to pull down your working branch to look at this.

**Comment by msmiths** [comment]
@MikeEdgar Yes... I already had most of @jmini's changes because I had applied all of the 2.0 related TODO's to my local branch a while ago. I have made that all of the changes from his branch have been applied.

The branch that I am using is: https://github.com/msmiths/smallrye-open-api/tree/mp-openapi-20

However, at the moment the JAX-RS module does not build because `GenericModelTypesResourceTest::testGenericsApplication` unit test is failing... as I mentioned on Gitter.
